lib/file-source/src/fingerprinter.rs

```diff
@@ -76,20 +77,12 @@ impl Fingerprinter {
             }
             FingerprintStrategy::Checksum {
                 ignored_header_bytes,
-                bytes,
-            } => {
-                buffer.resize(bytes, 0u8);
-                let mut fp = fs::File::open(path)?;
-                fp.seek(io::SeekFrom::Start(ignored_header_bytes as u64))?;
-                fp.read_exact(&mut buffer[..bytes])?;
-                let fingerprint = crc::crc64::checksum_ecma(&buffer[..]);
-                Ok(Checksum(fingerprint))
```

Comment:
This seems a bit confusing. Is this intended for a short transitional period? If not, we probably want to make the `FingerprintStrategy::Checksum` `unimplemented`, and rewrite the attempts to use it at the call sites (upper in the callstack). It will allow us to iteratively phase it out.

Reply:
Yes, this is a temporary state. Once we've released a version like this that knows how to do both, we can clean up the code to only do the one we want and instruct users that any migration path needs to go through the one with this upgrade mechanism.